src/index.ts

```diff
@@ -1583,8 +1582,26 @@ export class BigQuery extends common.Service {
       }
       // The Job is important for the `queryAsStream_` method, so a new query
       // isn't created each time results are polled for.
-      options = extend({job}, options);
-      job!.getQueryResults(options, callback as QueryRowsCallback);
+      job!
+        .on('error', err => callback!(err))
+        .on('complete', () => {
+          options = extend({job}, options);
+
+          if (
+            typeof query === 'string' &&
+            (query.includes('CREATE MODEL') || query.includes('REPLACE MODEL'))
+          ) {
+            job!.getQueryResults(options, callback as QueryRowsCallback);
```

Comment:
I think this method has polling logic built in, so we can move this check outside of the complete callback and return early.

Reply:
Gotcha